### PR TITLE
fix(typescript): make SetLabelOptions optional

### LINKED ISSUE #732: Storage: Support bucket.location_type
Location type is a new read-only field for buckets. This corresponds to internal bug 130187277.

### PR DESCRIPTION
**Pull request by callmehiphop**
test(bucket): location type system tests
docs(bucket): add verbiage about deprecated storage classes

---

Closes #732 

- [x] Tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

**Comment by callmehiphop** [comment]
@JesseLovelace as mentioned previously, no code change was required. So I've added some e2e tests to make sure the field was populated correctly.

File: src/bucket.ts [hunk 1]
```diff
@@ -2599,7 +2599,7 @@ class Bucket extends ServiceObject {
 
   setLabels(
     labels: Labels,
-    options: SetLabelsOptions
+    options?: SetLabelsOptions
```
Comment (JustinBeckwith) [inline src/bucket.ts]: Looks like this is a `fix` if we're making actual product changes, no?

Comment (callmehiphop) [reply src/bucket.ts]: Ahh, I forgot I tried to sneak that in 😱

File: src/bucket.ts [hunk 2]
```diff
@@ -2752,9 +2752,10 @@ class Bucket extends ServiceObject {
    *
    * @see [Storage Classes]{@link https://cloud.google.com/storage/docs/storage-classes}
    *
-   * @param {string} storageClass The new storage class. (`multi_regional`,
-   *     `regional`, `standard`, `nearline`, `coldline`, or
-   *     `durable_reduced_availability`)
+   * @param {string} storageClass The new storage class. (`standard`,
+   *     `nearline`, `coldline`, or `durable_reduced_availability`).
+   *     **Note:** The legacy storage classes `multi_regional` and `regional`
```
Comment (callmehiphop) [inline src/bucket.ts]: @JesseLovelace I thought about remove the available storage classes in favor of just linking to the upstream docs. I could see the list going stale and getting overlooked. WDYT?

Comment (JesseLovelace) [reply src/bucket.ts]: That's a valid concern, I'm good with either approach

**Pull request merged**